### PR TITLE
Add some docs content around enumeration attacks

### PR DESCRIPTION
**Pull request by whatyouhide**

### CODE REVIEW

File: lib/nimble_totp.ex [hunk 1]
```diff
@@ -85,23 +85,57 @@ defmodule NimbleTOTP do
...
+  a valid code from being reused](#module-preventing-codes-from-being-reused). However,
+  this approach by itself doesn't generally prevent an attacker from attempting to "guess"
+  a valid code by **enumerating** codes. TOTP codes are somewhat short, at only one million
+  possible values. You'll likely have other rate-limiting mechanisms in place that
+  would prevent an attacker from attempting codes in rapid sequence, but if you don't, you'll
+  also want to limit the number of attempts in some way.
+
+  For example, you can also store the last time a user attempted to enter *any* code, even
+  an invalid one. You can also use the `last_totp_at` value from the previous example to
+  store all attempts, not just valid codes. This way, you'll naturally prevent enumerating
```
Comment (josevalim) [inline lib/nimble_totp.ex:129]: I don't think will work. It means as soon as you put an empty value, you now have to wait 30 seconds (or the time it takes to expire).

File: lib/nimble_totp.ex [hunk 2]
```diff
@@ -85,23 +85,57 @@ defmodule NimbleTOTP do
...
+  If you only store the last time a user entered a *valid* TOTP code, you can [prevent
+  a valid code from being reused](#module-preventing-codes-from-being-reused). However,
+  this approach by itself doesn't generally prevent an attacker from attempting to "guess"
+  a valid code by **enumerating** codes. TOTP codes are somewhat short, at only one million
+  possible values. You'll likely have other rate-limiting mechanisms in place that
+  would prevent an attacker from attempting codes in rapid sequence, but if you don't, you'll
+  also want to limit the number of attempts in some way.
+
+  For example, you can also store the last time a user attempted to enter *any* code, even
+  an invalid one. You can also use the `last_totp_at` value from the previous example to
```
[the remark this answers is not in the document]
Comment (whatyouhide) [reply lib/nimble_totp.ex:128]: Yeah I agree. Maybe worth mentioning enumeration attacks and just take out this part?

[the remark this answers is not in the document]
Comment (josevalim) [reply lib/nimble_totp.ex:128]: Yes!!

[the remark this answers is not in the document]
Comment (whatyouhide) [reply lib/nimble_totp.ex]: Done.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**